utt/utt-client-api/src/User.cpp

```diff
@@ -447,9 +447,13 @@ void User::updateTransferTx(const Transaction& tx, const TxOutputSigs& sigs) {
 
     // Claim coins
     auto claimedCoins = pImpl_->client_->claimCoins(uttTx, pImpl_->params_, sigs);
-
+    bool invalidCoinsInTransfer(false);
     for (auto& coin : claimedCoins) {
-      if (!pImpl_->client_->validate(coin)) throw std::runtime_error("Invalid normal coin in transfer!");
+      if (!pImpl_->client_->validate(coin)) {
+        logdbg_user << "Invalid coin found; coin details: " << dbgPrintCoins({coin}) << endl;
```

Comment:
why not throw the exception here?
(and then you don't need  `invalidCoinsInTransfer`)

Reply:
The idea is to prevent prematurely exiting from the function. The recipient should be able to claim coins received after the invalid coin, so I am letting it process all coins before raising the exception.